src/proxy-ui-api/src/main/java/org/niis/xroad/restapi/openapi/TimestampingServicesApiController.java

```diff
@@ -62,9 +65,42 @@ public TimestampingServicesApiController(TimestampingServiceService timestamping
 
     @Override
     @PreAuthorize("hasAuthority('VIEW_TSPS')")
-    public ResponseEntity<List<TimestampingService>> getTimestampingServices() {
-        Collection<String> urls = timestampingServiceService.getTimestampingServices();
-        List<TimestampingService> timestampingServices = timestampingServiceConverter.convert(urls);
+    public ResponseEntity<List<TimestampingService>> getTimestampingServices(Boolean showConfigured) {
+        List<TimestampingService> timestampingServices;
+        // If show configured is true, return configured timestamping services from serverconf db.
+        // Otherwise return approved timestamping services from global conf.
+        if (showConfigured) {
+            List<TspType> tsps = timestampingServiceService.getConfiguredTimestampingServices();
+            timestampingServices = timestampingServiceConverter.convert(tsps);
+        } else {
+            Collection<String> urls = timestampingServiceService.getApprovedTimestampingServices();
+            timestampingServices = timestampingServiceConverter.convert(urls);
+        }
         return new ResponseEntity<>(timestampingServices, HttpStatus.OK);
     }
+
+    @Override
+    @PreAuthorize("hasAuthority('ADD_TSP')")
+    public ResponseEntity<TimestampingService> addTimestampingService(TimestampingService timestampingServiceToAdd) {
+        try {
+            timestampingServiceService.addConfiguredTimestampingService(timestampingServiceToAdd);
```

Comment:
I removed updated_at, message and status from the `TimestampingService` object that os used by `/system/timestamping-services`API since those attributes are related to the Diagnostics only. I created a new object `TimestampingServiceDiagnostics` that has those attributes. After the change the API returns a full object with all the attributes (`name`, `url`) populated.

Reply:
Nice :+1: